src/main/java/jchess/gamelogic/Game.java

```diff
@@ -101,6 +101,9 @@ public void onTimerElapsed() {
 	public void onBoardActivity(IPositionAgent oPosition) {
 		m_oLogger.writeLog(LogLevel.DETAILED, "Activity on board has been observed.", "onBoardActivity", "Game");
 
+		// This checks the stalemate rule
+		checkStalemate();
```

Comment:
This return does not return any value. Mean what if Stalemate is true should not we stop the game? or can use continue with the game...

Reply:
we will stop the game. The function for the same is being developed by Danny so will reuse the same function.